src/app/clusters/joint-fabric-administrator-server/joint-fabric-administrator-server.cpp

```diff
@@ -102,19 +102,48 @@ void MatterJointFabricAdministratorPluginServerInitCallback()
     AttributeAccessInterfaceRegistry::Instance().Register(&gJointFabricAdministratorAttrAccess);
 }
 
-// TODO
+#if CHIP_DEVICE_CONFIG_ENABLE_JOINT_FABRIC
+bool __attribute__((weak)) emberAfJointFabricAdministratorClusterGetIcacCsr(MutableByteSpan & icacCsr)
+{
+    return false;
+}
+
 bool emberAfJointFabricAdministratorClusterICACCSRRequestCallback(
     chip::app::CommandHandler * commandObj, const chip::app::ConcreteCommandPath & commandPath,
     const chip::app::Clusters::JointFabricAdministrator::Commands::ICACCSRRequest::DecodableType & commandData)
 {
     MATTER_TRACE_SCOPE("ICACCSRRequest", "JointFabricAdministrator");
 
-    ChipLogProgress(Zcl, "JointFabricAdministrator: Received a ICACCSRRequest command");
+    ChipLogProgress(Zcl, "JointFabricAdministrator: Received an ICACCSRRequest command");
+
+    auto nonDefaultStatus = Status::Success;
+    uint8_t buf[Credentials::kMaxDERCertLength];
+    MutableByteSpan icacCsr(buf, Credentials::kMaxDERCertLength);
+    Commands::ICACCSRResponse::Type response;
+
+    auto & failSafeContext = Server::GetInstance().GetFailSafeContext();
+    VerifyOrExit(failSafeContext.IsFailSafeArmed(commandObj->GetAccessingFabricIndex()),
+                 nonDefaultStatus = Status::FailsafeRequired);
+
+    /* TODO spec.: If this command is received from a peer against FabricFabric Table Vendor ID Verification
```

Comment:
This sentence (both here and in the spec) has grammar issues....

Reply:
fixed the phrasing both in the [spec ](https://github.com/CHIP-Specifications/connectedhomeip-spec/pull/11848/files)and in this PR: 

`If the <<ref_FabricTableVendorIdVerificationProcedure, FabricFabric Table Vendor ID Verification Procedure>> has not been executed against the initiator of this command, the command SHALL fail with a <<ref_JFVidNotVerified, JfVidNotVerified>> status code SHALL be sent back to the initiator`